src/main/java/singly_linked_list/ListService.java

```diff
@@ -1,33 +1,216 @@
+            cur.info = info;        //cur.setInfo(info)
+            cur.next = top;
+            top = cur;
+            info = scanner.nextInt();
+        }
+        return top;
     }
 
+
+    @Override
     public Node createFifo() {
-        return null;
+        Node top = null, cur;
+        int num = scanner.nextInt();
+        if(num != 0){
+            top = new Node();
+            top.info = num;
+            top.next = null;
+            cur = top;
+            num = scanner.nextInt();
+            while(num != 0){
+                cur.next = new Node();
+                cur = cur.next;
+                cur.info = num;
+                num = scanner.nextInt();
+            }
+            cur.next = null;
+        }
+        return top;
     }
 
-    public void printList(Node top) {
 
+    @Override
+    public void printList(Node top) {
+        if(top == null){
+            System.out.println("The list is empty.");
+            return;
+        }
+        while(top != null){
+            System.out.print(top.info + " ");
+            top = top.next;
+        }
+        System.out.println("\n");
     }
 
-    public void printListRecursively(Node top) {
 
+    @Override
+    public void printListRecursively(Node top) {
+        if(top == null){
+            System.out.print("\n");
+            return;
+        }
+        System.out.print(top.info + " ");
+        printListRecursively(top.next);
     }
 
-    public Node remove(Node top, int toRemove) {
+
+    @Override
+    public Node remove(Node top, int toDelete) {
+        Node cur;
+        Node prev;
+        if(top == null){
+            return null;
+        }
+        if(toDelete == top.info){
+            cur = top;
+            top = top.next;
+            cur = null;
+            return top;
+        }
+        prev = top;
+        cur = top.next;
+        while(cur != null){
+            if(cur.info == toDelete){
+                prev.next = cur.next;
+                cur = null;
+                return top;
+            }
+            prev = cur;
+            cur = cur.next;
+        }
         return null;
     }
 
-    public void insertBefore(Node top, int toInsert, int elementBefore) {
+    @Override
+    public Node insertBefore(Node top, int toAdd, int before) {
+        Node cur = top.next;
+        Node prev = top;
+        Node additional = new Node();
+        additional.info = toAdd;
+        if(top.info == before){
+            top = additional;
+            top.next = prev;
+            return top;
+        }
+        while(cur != null){
+            if(cur.info == before) {
+                prev.next = additional;
+                additional.next = cur;
+                return top;
+            }
+            prev = cur;
+            cur = cur.next;
+        }
+        System.out.println("There is no such number (" + before +
+                ") in the sequence before which you want to put a number " + toAdd);
+        return top;
```

Comment:
Please remove using of ```System.out.println()``` here and throw an exception

Reply:
Corrected.